Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/conceptcollection/ConceptCollectionManagerTest.java

```diff
@@ -0,0 +1,219 @@
+import edu.asu.spring.quadriga.domain.conceptcollection.IConcept;
+import edu.asu.spring.quadriga.domain.conceptcollection.IConceptCollection;
+import edu.asu.spring.quadriga.domain.conceptcollection.IConceptCollectionCollaborator;
+import edu.asu.spring.quadriga.domain.factory.conceptcollection.IConceptFactory;
+import edu.asu.spring.quadriga.domain.impl.ConceptpowerReply;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.Concept;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.ConceptCollection;
+import edu.asu.spring.quadriga.domain.impl.conceptcollection.ConceptCollectionCollaborator;
+import edu.asu.spring.quadriga.domain.impl.workbench.Project;
+import edu.asu.spring.quadriga.domain.impl.workspace.WorkSpace;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.domain.workspace.IWorkSpace;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.conceptcollection.mapper.IConceptCollectionDeepMapper;
+import edu.asu.spring.quadriga.service.workbench.mapper.IProjectShallowMapper;
+import edu.asu.spring.quadriga.service.workspace.IListWSManager;
+import mockit.Mocked;
+
+public class ConceptCollectionManagerTest {
+
+    @Mock
+    private RestTemplate mockedRestTemplate = Mockito.mock(RestTemplate.class);
+
+    @Mock
+    private IConceptCollectionDeepMapper mockedConceptCollectionDeepMapper = Mockito
+            .mock(IConceptCollectionDeepMapper.class);
+
+    @Mock
+    private IProjectShallowMapper mockedProjectShallowMapper = Mockito.mock(IProjectShallowMapper.class);
+
+    @Mock
+    private IListWsDAO mockedwsListManger = Mockito.mock(IListWsDAO.class);
+
+    @Mock
+    private IListWSManager mockedwsManager = Mockito.mock(IListWSManager.class);
+
+    @Mock
+    private IConceptFactory mockedConceptFactory = Mockito.mock(IConceptFactory.class);
+
+    @Mock
+    private IConceptCollectionDAO mockedccDao = Mockito.mock(IConceptCollectionDAO.class);
+
+    @Mocked("conceptURL")
+    private String conceptURL;
+
+    @Mocked("searchURL")
+    private String searchURL;
+
+    @InjectMocks
+    private ConceptCollectionManager conceptCollectionManagerUnderTest;
+
+    private Concept concept;
+
+    @Before
+    public void setUp() throws QuadrigaStorageException {
+        MockitoAnnotations.initMocks(this);
+
+        List<ConceptpowerReply.ConceptEntry> conceptEntries = new ArrayList<ConceptpowerReply.ConceptEntry>();
+        ConceptpowerReply.ConceptEntry entry = new ConceptpowerReply.ConceptEntry();
+        entry.setLemma("test-lemma");
+        conceptEntries.add(entry);
+        ConceptpowerReply rep = new ConceptpowerReply();
+        rep.setConceptEntry(conceptEntries);
+
+        Mockito.when(mockedRestTemplate.getForObject(Matchers.eq(conceptURL + searchURL + "{name}/{pos}"),
+                Matchers.eq(ConceptpowerReply.class), Matchers.anyMapOf(String.class, String.class))).thenReturn(rep);

```

Comment:
actually, the third argument is where you need to provide id/name/pos so that the rest template returns the correct answer for the given name/pos or id. sorry, I didn't look closely enough before.